shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -236,8 +236,18 @@ public void stopRenderingToSurface() {
     isDisplayingFlutterUi = false;
   }
 
-  // TODO(mattcarroll): describe the native behavior that this invokes
+  /**
+   * Notifies Flutter that the viewport metrics, e.g. window height and width, have changed.
+   *
+   * <p>If the width, height, or devicePixelRatio are less than or equal to 0, this update is
+   * ignored.
+   *
+   * @param viewportMetrics The metrics to send to the Dart application.
+   */
   public void setViewportMetrics(@NonNull ViewportMetrics viewportMetrics) {
+    if (!viewportMetrics.validate()) {
```

Comment:
what's the edge case here? Leave a code comment for the next person?

Reply:
also, should it throw?

Reply:
We call this method when we get DPR updates, which can happen before we get width/height. The embedding ends up calling this with just the DPR sometimes.

Reply:
Added a comment